frontend/src/metabase/query_builder/components/view/PreviewQueryModal/PreviewQueryModal.unit.spec.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import xhrMock from "xhr-mock";
+import { render, screen } from "@testing-library/react";
+import Question from "metabase-lib/Question";
+import PreviewQueryModal from "./PreviewQueryModal";
+
+const SQL_QUERY = "SELECT 1";
+const SQL_QUERY_ERROR = "Cannot run the query";
+
+describe("PreviewQueryModal", () => {
+  beforeEach(() => {
+    xhrMock.setup();
+  });
+
+  afterEach(() => {
+    xhrMock.teardown();
+  });
+
+  it("should show a fully parameterized native query", async () => {
+    const question = Question.create({ databaseId: 1 });
+    mockNativeQuery();
+
+    render(<PreviewQueryModal question={question} />);
+
+    expect(await screen.findByText(SQL_QUERY)).toBeInTheDocument();
```

Comment:
I find expecting the `SQL_QUERY` part a bit confusing without looking into how `mockNativeQuery` is implemented. I think it'd be more apparent if we are to pass `SQL_QUERY` down to `mockNativeQuery` so that when we read the test case we know we mock the API to return the expected SQL query so we should be able to find it on the screen.

The same could be said for `mockNativeQueryError` and `SQL_QUERY_ERROR`

What do you think?